src/MongoDB.Driver/Support/ReflectionExtensions.cs

```diff
@@ -89,6 +89,19 @@ public static bool IsConvertibleToEnum(this Type type)
                 type == typeof(string);
         }
 
+        public static bool IsNumericUnderlyingEnumType(this Type type)
```

Comment:
This new method is potentially not actually needed depending on how you resolve the comment I made on the line that calls this method.